frontend/src/app/components/user-access/user-access.component.ts

```diff
@@ -29,14 +34,52 @@ export class UserAccessComponent implements OnInit {
   }
 
   login(): void {
-    this.backendService.login(this.user)
-      .subscribe((user) => {
-        this.toastService.loginSuccessful()
-        this.localStorage.setLoggedStatus()
-        this.showDialog(false)
-      }, (error) => {
-        console.log(error)
-        this.toastService.error()
-      })
+    this.resetErrors();
+
+    if (this.inputAccepted()) {
+      this.backendService.login(this.user)
+        .subscribe(() => {
+          this.accessSuccessful = true;
+
+          this.toastService.loginSuccessful();
+          this.localStorage.setLoggedStatus();
+          this.showDialog(false);
+        }, (error) => {
+          if (error && (error.status === 401 || error.status === 406)) {
+            this.toastService.loginFailed();
+          } else {
+            this.toastService.error();
+          }
+
+          this.accessError = true;
+          console.log(error);
+        });
+    }
+  }
+
+  private inputAccepted(): boolean {
```

Comment:
Da aggiungere il controllo nel caso in cui username e password siano vuoti: la variabile è undefined inizialmente ma se si scrive qualcosa e poi si cancella la variabile risulta essere una stringa vuota